### PR TITLE
Sigma derivative fix

### LINKED ISSUE #550: Sigma derivative in QPfunc with extra offset
The derivative of the self-energy in `GW::QPfunc` is returned as 

```      
return _sigma_c_func.CalcCorrelationDiagElementDerivative(_gw_level,
                                                                frequency)+
             _offset - frequency;
```

where offset is the intercept `frequency + sigma_x - Vxc` so the function effectively returns `dSigma/dw = dSigma_c/dw + Sigma_x - Vxc`

That is just a constant but shouldn't be here. Unless I understand this wrong.

### PR DESCRIPTION
**Pull request by baumeier**
Fixes #550

### CODE REVIEW

**Review by JensWehner — APPROVED** [review]
Yes that was a bug. In the combined version that is not done... hmm.

**Review by JensWehner — CHANGES_REQUESTED** [review]
actually it misses a -1.0

File: include/votca/xtp/gw.h [hunk 1]
```diff
@@ -125,8 +125,7 @@ class GW {
     }
     double deriv(double frequency) const {
       return _sigma_c_func.CalcCorrelationDiagElementDerivative(_gw_level,
-                                                                frequency) +
-             _offset - frequency;
+                                                                frequency);
```
Comment (JensWehner) [inline include/votca/xtp/gw.h:128]:
```suggestion
                                                                frequency)-1.0; // -1.0 comes from derivative of the frequency itself.
```

**Review by JensWehner — APPROVED** [review]

File: include/votca/xtp/gw.h (hunk 1 above)
Comment (baumeier) [reply include/votca/xtp/gw.h]: How? This is just the freq derivative of Sigma?

Comment (baumeier) [reply include/votca/xtp/gw.h]: ok, this is used in the grid solver for the root finding problem, when `f(w)=Sigma_c(w) + w0 + Sigma_x - Vxc - w = 0`

But then, the `qpweight` is calculated from this as as `1/(1 - f'(w)) = 1/(1-Sigma_c(w)' +1)`, no? That should be correcte to `1/(-f'(w))`, I guess.

Comment (JensWehner) [reply include/votca/xtp/gw.h]: I do not remember what the quasiparticle weight definition is.

Comment (baumeier) [reply include/votca/xtp/gw.h]: `1/(1-Sigma')`

Comment (JensWehner) [reply include/votca/xtp/gw.h]: true.

Comment (JensWehner) [reply include/votca/xtp/gw.h]: then you mention is correct.

File: include/votca/xtp/gw.h [hunk 2]
```diff
@@ -111,7 +111,12 @@ class GW {
         : _gw_level(gw_level), _offset(offset), _sigma_c_func(sigma){};
     std::pair<double, double> operator()(double frequency) const {
       std::pair<double, double> value;
-      value.first = value(frequency) value.second = deriv(frequency);
+      value.first =
+          _sigma_c_func.CalcCorrelationDiagElement(_gw_level, frequency);
+      value.second = _sigma_c_func.CalcCorrelationDiagElementDerivative(
+          _gw_level, frequency);
+      value.first += (_offset - frequency);
+      value.second -= 1.0;
```
Comment (JensWehner) [inline include/votca/xtp/gw.h]: why this? The old stuff was missing a `;` but otherwise it should just call the individual functions.

**Review by JensWehner — CHANGES_REQUESTED** [review]
why not use the `value` and `deriv` function in the `()` operator?

File: include/votca/xtp/gw.h (hunk 2 above)
Comment (baumeier) [reply include/votca/xtp/gw.h]: I didn't realize that it had failed thanks to double use of `value`. Changed the return variable name.

File: include/votca/xtp/gw.h [hunk 3]
```diff
@@ -110,23 +110,26 @@ class GW {
...
-      value.first += (_offset - frequency);
-      value.second -= 1.0;
-      return value;
+      std::pair<double, double> result;
+      // value.first =
+      //    _sigma_c_func.CalcCorrelationDiagElement(_gw_level, frequency);
+      // value.second = _sigma_c_func.CalcCorrelationDiagElementDerivative(
+      //    _gw_level, frequency);
+      // value.first += (_offset - frequency);
+      // value.second -= 1.0;
```
Comment (JensWehner) [inline include/votca/xtp/gw.h:119]: these comments should still go

Comment (baumeier) [reply include/votca/xtp/gw.h:119]: argh

**Review by JensWehner — APPROVED** [review]

**Pull request merged**